src/viam/sdk/components/camera/client.cpp

```diff
@@ -22,84 +23,44 @@ CameraClient::CameraClient(std::string name, std::shared_ptr<grpc::Channel> chan
       channel_(std::move(channel)){};
 
 AttributeMap CameraClient::do_command(AttributeMap command) {
-    viam::common::v1::DoCommandRequest req;
-    viam::common::v1::DoCommandResponse resp;
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    const google::protobuf::Struct proto_command = map_to_struct(command);
-    *req.mutable_command() = proto_command;
-    *req.mutable_name() = this->name();
-    stub_->DoCommand(&ctx, req, &resp);
-    return struct_to_map(resp.result());
+    return make_client_helper(this, *stub_, &StubType::DoCommand)
+        .with([&](auto& request) { *request.mutable_command() = map_to_struct(command); })
+        .invoke([](auto& response) { return struct_to_map(response.result()); });
 };
 
 Camera::raw_image CameraClient::get_image(std::string mime_type, const AttributeMap& extra) {
-    viam::component::camera::v1::GetImageRequest req;
-    viam::component::camera::v1::GetImageResponse resp;
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    Camera::normalize_mime_type(mime_type);
-
-    *req.mutable_mime_type() = mime_type;
-    *req.mutable_name() = this->name();
-    *req.mutable_extra() = map_to_struct(extra);
-
-    stub_->GetImage(&ctx, req, &resp);
-    return from_proto(resp);
+    return make_client_helper(this, *stub_, &StubType::GetImage)
+        .with(extra,
+              [&](auto& request) {
+                  // CR erodkin: call attention to flyby here
+                  *request.mutable_mime_type() = Camera::normalize_mime_type(mime_type);
```

Comment:
(flyby) previously we weren't using the return from the `normalize_mime_type` call so the request's mime type wasn't actually being normalized.